test/test_util.jl

```diff
@@ -20,16 +20,28 @@ at input point `x` to confirm that there are correct ChainRules provided.
 All keyword arguments except for `fdm` and `test_wirtinger` are passed to `isapprox`.
 """
 function test_scalar(f, x; rtol=1e-9, atol=1e-9, fdm=_fdm, test_wirtinger=x isa Complex, kwargs...)
+    ensure_not_running_on_functor(f, "test_scalar")
+
     @testset "$f at $x, $(nameof(rule))" for rule in (rrule, frule)
         res = rule(f, x)
         @test res !== nothing  # Check the rule was defined
-        fx, ∂x = res
+        fx,  prop_rule = res
         @test fx == f(x)  # Check we still get the normal value, right
 
+        if rule == rrule
+            ∂self, ∂x = prop_rule(1)
+            @test ∂self === NO_FIELDS
+        else # rule == frule
+            # Got to input extra first aguement for internals
+            # But it is only a dummy since this is not a functor
+            ∂x, = prop_rule(NamedTuple(), 1)
```

Comment:
this is only right for `scalar_rules`, right? Which is fine, given the context, but i'd maybe still like a comment saying that

Reply:
What is only right for `scalar_rules`?

Reply:
because otherwise the first result is not `∂x` but `∂self`? 

i.e. it is the corollary of this comment
https://github.com/JuliaDiff/ChainRulesCore.jl/blob/d7a370f68bcf8e9ec2931beae3398038f4388a4d/src/rule_definition_tools.jl#L212-L216

Reply:
Nope, for `frule` you don't get back a `∂self`,
instead you need to put in `Δself`

Reply:
in which case I do not understand the code and/or comment I linked...

Reply:
oh, is it that
for the function `self(ws...) -> x::Number`
the "pushforward" for `self` takes two args `Δself, Δws` (in the `scalar_rule` case `Δself` is ignored) and returns one output `∂x`

specifically 
```julia
fws::Number,  pushforward::Function = frule(self, ws) 
∂x::Number, = pushforward(anything, x)::Tuple{Any}
```

Reply:
yeah